### ISSUE TITLE
Unexpected return value when expression uses an array length in a function

### ISSUE THREAD

**Issue by NoOverflow** [n8n-io/n8n#5226]
Hi, I'm not sure if this is a duplicate (I've looked into open issues but couldn't find a close match), or if this issue comes from a misunderstanding on how to use n8n but I still find it weird.

**Describe the bug**

When using the length of an array set in a previous node, in a function (such as Math.floor), in an inline expression, the return value will always be equal to the actual length no matter what the expression is supposed to do.  Hardcoding the length results in expected behaviour.

**To Reproduce**

(Example workflow provided as attachment for convenience)
1. Create a workflow with an array set as test value named `my_array`
2. Create a "Set" node that set a variable called index from the following expression: 
```javascript
{{ Math.floor(Math.random() * $json["my_array"].length) }}
```
3. Confirm that returned value is always equals to the array length and not what actual JS execution would return
4. Create another "Set" node that swap the dynamic array length for an hardcoded value using the following expression: 
```javascript
{{ Math.floor(Math.random() * 6) }}
```
5. Confirm that the returned value matches what would be expected for this expression computation

**Expected behavior**

An expression should be able to contain the dynamic length of an array

**Environment (please complete the following information):**

- OS: Tested on Ubuntu Linux 22.10, Centos 7
- n8n Version 0.212.0 (Latest on docker)
- Node.js Version (Matching docker latest)
- Database system SQLite
- Operation mode own
- Tested on both Docker install and Openshift

**Additional context**
I attached the test workflow for convenience, however it's lacking the n8n test data, you can just add a dummy array containing a few values.

[workflow.zip](https://github.com/n8n-io/n8n/files/10480612/workflow.zip)

**Comment by NoOverflow** [comment]
Posting additional screenshot that helps understand the issue (Return value here should be 800 + 6 = 806)
![image](https://user-images.githubusercontent.com/54811870/214065789-6cba081f-aacd-4a0f-9f5a-2f9e5618227b.png)

**Comment by Joffcom** [comment]
Hey @NoOverflow,

Thanks for reporting this one, I must be missing something as I can see the result is wrong in your screenshot but when I run a workflow to try and reproduce it is running as expected.

...

As I was about to post this I just noticed I was using `.length()` and that was working but `.length` is failing. 

![image](https://user-images.githubusercontent.com/4688521/214069623-145db223-f64d-41df-90a2-5ee714b31891.png)

I have created `N8N-6026` as our internal ticket for tracking this one.
